### DISCUSSION TITLE
Could you please upgrade the SponsorBlock fork ?

### DISCUSSION THREAD

**Discussion by KaKi87** [TeamNewPipe/NewPipe#10713] [Questions] [answered]
Hello,

@polymorphicshade, @Stypox, @gilbsgilbs,

The SponsorBlock fork at [gilbsgilbs/NewPipeSponsorBlock](https://github.com/gilbsgilbs/NewPipeSponsorBlock) is currently stuck at NewPipe v0.25.2, would it be possible to upgrade it to the latest version (v0.26.1) so we could enjoy the new features of this minor version ?

Thanks

**Comment by gilbsgilbs** [comment, marked as answer]
The actual GitHub repo of NewPipeSponsorBlock is: https://github.com/polymorphicshade/NewPipe . IIUC, this fork is no longer maintained (see https://github.com/polymorphicshade/NewPipe/discussions/260), and this project is not officially maintained or endorsed by the NewPipe team.

**Reply by KaKi87** [reply]
> The actual GitHub repo of NewPipeSponsorBlock is: polymorphicshade/NewPipe . IIUC, this fork is no longer maintained (see polymorphicshade#260)

Is *Tubular* a new project that is maintained, and if yes, where ?

> this project is not officially maintained or endorsed by the NewPipe team.

I know, but there were no other place where I could ask.

Thanks

**Reply by TheAssassin** [reply]
> I know, but there were no other place where I could ask.

That claim is wrong. See https://github.com/TeamNewPipe/NewPipe/discussions/10713#discussioncomment-7977883.

**Comment by TheAssassin** [comment]
NewPipe does not endorse nor support any of these forks. They are forks, so by definition, the people who forked are responsible. This is the wrong place to ask for support.

**Reply by KaKi87** [reply]
I know, but there were no other place where I could ask.

Thanks

**Reply by TheAssassin** [reply]
Still, this is not the place. The maintainer has an email address on their GitHub profile. You can use that.

**Reply by KaKi87** [reply]
No dev answers support requests by e-mail, and even if they did, it would only benefit me, while many other people could be wondering and therefore find this post and get an answer from it, also the maintainer wouldn't be happy either if every person wondering would send an e-mail.

**Reply by TheAssassin** [reply]
Quite a bold claim. I often do. Many others often do. It's the reason we publish the address in the first place.

The maintainer needs to publish a statement. They can and should not do so here because this is still the wrong place. They have their own repository. Someone needs to tell them, though.

Your argument does not make any sense from the perspective of an unrelated project. You could post your discussion in youtube-dl's tracker as well, but it would make just as little sense as it does here.

Your question has been answered twice. Please contact the upstream project. We told you this is the wrong place to discuss that matter. You should respect that.

**Reply by KaKi87** [reply]
> It's the reason we publish the address in the first place.

Usually, the reason why devs publish an adress isn't that at all, but rather to receive job offers.

> You should respect that.

Like you respect that at least dozens, in reality probably hundreds, of NewPipe users, would like SponsorBlock to be integrated to it ?

**Reply by TheAssassin** [reply]
This is entirely off topic now. But I'd like to provide a final response.

The audacity of your statement really is stunning. You are referring to respect? Well, you just showed that _you_ have no respect. Do you even realize that the developers develop NewPipe in their free time without ever receiving any financial benefit? That they make the resulting code available for free? That they build binaries and publish them not only on GitHub but also on F-Droid (both own repository and upstream)? That they spend time building new features and supporting people? And all of that free of charge?

Plenty other tools based on NewPipe and even those forks benefit from the work, they would not be possible if NewPipe hadn't been an open-source project. Which means you couldn't even complain now about the lack of updates of that fork here, because it just wouldn't exist _at all_.

Every developer of any application chooses themselves what they work on. Users may _suggest_ features, but they cannot _demand_ them. Please remember these words the next time you request a feature somewhere and think twice about your wording.

By the way, the team has never had any argument with @polymorphicshade and others forking NewPipe. Every interaction I ever saw was done respectfully from both sides. It's the users that are angry and attack the project.

In any case, there is no reason to continue here. Question answered. Discussion closed. We may lock it in the future.

**Reply by KaKi87** [reply]
> Do you even realize that the developers develop NewPipe in their free time without ever receiving any financial benefit? That they make the resulting code available for free?

I do, and that's why I find unacceaptable that you refused to merge SponsorBlock support that someone did took the time to implement for free.

**Reply by opusforlife2** [reply]
That is some very strange logic. You seem to be coming at this from two extremes, one being of entitlement ("gib feature ASAP"), and the other of a misplaced sense of justice ("they put in effort so you're _obligated_ to merge it").

No developer is obligated to accept _any_ contribution. That is the entire reason forks exist. Just to take an extreme hypothetical example, if someone spent three years of hard work on a PR to add user tracking and analytics to Newpipe, should we merge it in order to "respect" the PR author's efforts?
